### DISCUSSION TITLE
tinydb-serialization and documents

### DISCUSSION THREAD

**Discussion by jcphlux** [msiemens/tinydb#475] [Q&A]
Can [tinydb-serialization](https://github.com/msiemens/tinydb-serialization) work for documents or only properties of the documents?
We have classes for all our data and it would be nice if I did not have to convert all the return values to classes.

**Comment by msiemens** [comment]
Am I understanding correctly that the issue you're having is that you have a lot of classes and don't want to write and register a custom serializer for each one? If that is the case you're right that at the moment one serializer class only can write/read a single class. But I think we could redesign it to allow a single serializer to support multiple classes. If I'm not understanding correctly, feel free to clear up any confusions :)

**Reply by jcphlux** [reply]
Maybe the way I am reading the code is wrong but it looks like tinydb-serialization is for fields and not documents.
```python
# this
db.insert({'custom_field': custom_field})
# not this
db.insert(custom_class)
```
so maybe it is a different ask to for it to work with classes.
The insert is not that big a deal it is more the extra steps of having to cast on read.

**Reply by msiemens** [reply]
Sorry for the late reply! You're right, I misunderstood and now see what you mean 🙂 I think that's a good idea, but I just found that right now the main obstacle for implementing this is that TinyDB itself will complain if you pass a document that is not a `dict` or `dict`-like. TinyDB-Serialization can't help here as TinyDB performs this check before the middleware gets involved. I'll have to think about if and how this could be solved.
